tools/sil-llvm-gen/SILLLVMGen.cpp

```diff
@@ -184,26 +184,28 @@ int main(int argc, char **argv) {
   if (CI.getASTContext().hadError())
     return 1;
 
-  // Load the SIL if we have a module. We have to do this after SILParse
-  // creating the unfortunate double if statement.
-  if (Invocation.hasSerializedAST()) {
-    assert(!CI.hasSILModule() &&
-           "performSema() should not create a SILModule.");
-    CI.createSILModule();
-    std::unique_ptr<SerializedSILLoader> SL = SerializedSILLoader::create(
-        CI.getASTContext(), CI.getSILModule(), nullptr);
-
-    if (extendedInfo.isSIB())
-      SL->getAllForModule(CI.getMainModule()->getName(), nullptr);
-    else
-      SL->getAll();
+  auto *mod = CI.getMainModule();
+  assert(mod->getFiles().size() == 1);
+
+  std::unique_ptr<SILModule> SILMod;
+  if (PerformWMO) {
+    SILMod = performSILGeneration(mod, CI.getSILTypes(), CI.getSILOptions());
```

Comment:
Can we use a different name than performSILGeneration. It makes it sound like you are SILGening code. It should be really clear that we are parsing SIL from the name of this method. One thought is imagine if we had a helper method that called performSILGeneration internally and takes the compiler instance, but with a name like performSILParsing?

Reply:
Also, while you are touching this code, do you think we can centralize any of this code that is in the various tools into helper code rather than duplicating?

Reply:
If this doesn't actually need to perform code generation (is there a canonicalization step that needs to get done here?), an entrypoint to ParseSILModuleRequest, or even invoking it directly would work here.

Reply:
Hmm, we can't call into `ParseSILModuleRequest` directly (at least not without adding a special case for when we have a .sil file input), as we also need to be handle SIB.

My goal with this was for `performSILGeneration` to be the one function you call to turn a `ModuleDecl *` or `FileUnit *` into a `SILModule`, regardless of specifics of what the AST nodes contain. In the future, it might perform both SIL parsing and SIL generation if we allow SIL files to mix with Swift files.

Perhaps a better name would be something more general like `lowerAST` or `performASTLowering`?